### ISSUE TITLE
Tabufline crash

### ISSUE THREAD

**Issue by KorigamiK** [NvChad/ui#106]
This crash happens whenever I try to open a file in a tab from the file tree by `Ctrl` + `t`

```bash
...l/share/nvim/lazy/ui/lua/nvchad_ui/tabufline/modules.lua:209: in function <...l/share/nvim/lazy/ui/lua/nvchad_ui/tabufline/modules.lua:201>
   E15: Invalid expression: v:lua.require('nvchad_ui.tabufline.modules').run()
ENTER or type command to continue
```

**Comment by siduck** [comment]
duplicate of https://github.com/NvChad/ui/issues/63 , this would happen only straight after installation, like if you try to use nvim straight after installing, reopen nvim and the issue wont be there!


anyways i'll look into it tomorrow

**Comment by KorigamiK** [comment]
Hi, as it turns out, this issue persists after installation. In fact, the first time installing the config didn't give any errors and the tabufline worked fine.

For now, pressing tab or shift tab doesn't do anything and opening the buffer in a new tab crashes tabufline.

My configuration is here https://github.com/KorigamiK/dotfiles/tree/new-nvim/.config/nvim

https://github.com/NvChad/ui/assets/72932688/ce0b5384-747d-49ea-9afa-23b011e3749c

**Comment by siduck** [comment]
hi @KorigamiK can you show me output of :mes, command when you have the error?

**Comment by KorigamiK** [comment]
Sure, this is the output of `:mes` after getting the error

```bash
^I...l/share/nvim/lazy/ui/lua/nvch...chad_ui/tabufline/modules.lua:201>
E15: Invalid expression: v:lua.require('nvchad_ui.tabufline.modules').run()
:mes
...l/share/nvim/lazy/ui/lua/nvchad_ui/tabufline/modules.lua:155: bad argument #1 to
 'ipairs' (table expected, got nil)
stack traceback:
^I[C]: in function 'ipairs'
^I...l/share/nvim/lazy/ui/lua/nvchad_ui/tabufline/modules.lua:155: in function 'buf
ferlist'
^I...l/share/nvim/lazy/ui/lua/nvchad_ui/tabufline/modules.lua:209: in function <...
l/share/nvim/lazy/ui/lua/nvchad_ui/tabufline/modules.lua:201>
E15: Invalid expression: v:lua.require('nvchad_ui.tabufline.modules').run()
Press ENTER or type command to continue
```

**Comment by KorigamiK** [comment]
So, I tried to add this line before  passing `vim.t.bufs` and the tabs seem to load now (I don't know if they are correctly working), but The buffers are still not updated on opening new buffers or jumping through them.

```lua
  if vim.t.bufs then
    for _, bufnr in ipairs(vim.t.bufs) do
    -- rest of the code
```

**Comment by siduck** [comment]
> Sure, this is the output of `:mes` after getting the error
> 
> ```shell
> ^I...l/share/nvim/lazy/ui/lua/nvch...chad_ui/tabufline/modules.lua:201>
> E15: Invalid expression: v:lua.require('nvchad_ui.tabufline.modules').run()
> :mes
> ...l/share/nvim/lazy/ui/lua/nvchad_ui/tabufline/modules.lua:155: bad argument #1 to
>  'ipairs' (table expected, got nil)
> stack traceback:
> ^I[C]: in function 'ipairs'
> ^I...l/share/nvim/lazy/ui/lua/nvchad_ui/tabufline/modules.lua:155: in function 'buf
> ferlist'
> ^I...l/share/nvim/lazy/ui/lua/nvchad_ui/tabufline/modules.lua:209: in function <...
> l/share/nvim/lazy/ui/lua/nvchad_ui/tabufline/modules.lua:201>
> E15: Invalid expression: v:lua.require('nvchad_ui.tabufline.modules').run()
> Press ENTER or type command to continue
> ```

cool, will try to fix

**Comment by ariel-frischer** [comment]
Same error, how can I disable tabufline?

**Comment by ariel-frischer** [comment]
Got it in M.ui nvchad config set enabled to `false`:
```
  tabufline = {
    enabled = false,
  },
```
This was really breaking my auto-session manager so had to disable for now.

**Comment by ariel-frischer** [comment]
Thank you @siduck  for the super quick bugfix, I've done :Lazy sync and its working again!

**Comment by KorigamiK** [comment]
Hi, I did pull the latest changes as well, but I don't seem to have any luck.
The same error still persists. When I open any file manually or through nvim-tree it isn't updated to the buffers and creating a tab causes the crash and so does opening neogit.

I have my entire custom config uploaded to [this branch here](https://github.com/KorigamiK/dotfiles/tree/new-nvim/.config/nvim)

To be honest, I really like this config but not tabbufline not working makes it very hard to use. I hope you can look into this one more time @siduck

**Comment by siduck** [comment]
> Hi, I did pull the latest changes as well, but I don't seem to have any luck. The same error still persists. When I open any file manually or through nvim-tree it isn't updated to the buffers and creating a tab causes the crash and so does opening neogit.
> 
> I have my entire custom config uploaded to [this branch here](https://github.com/KorigamiK/dotfiles/tree/new-nvim/.config/nvim)
> 
> To be honest, I really like this config but not tabbufline not working makes it very hard to use. I hope you can look into this one more time @siduck

reopen neovim after doing lazy sync please, neogit works fine here now

![image](https://github.com/NvChad/ui/assets/59060246/bfb6e684-cdb6-450f-94b6-962b88cc09a8)

**Comment by KorigamiK** [comment]
Guess what @siduck, I found the issue. It's these innocent looking 4 lines which went unnoticed:

```lua
-- Highlight on yank
vim.cmd([[
  autocmd!
  autocmd TextYankPost * silent! lua vim.highlight.on_yank()
  augroup end
]])
```

I don't know why but removing them makes everything work magically! :rocket: :partying_face: 

But I'm glad we were able to identify and fix other possible issues due to this :laughing: 
Although, I would like to know how to get highlight on yank working again.
